Xledger.Sql/Collections/ImmList.cs

```diff
@@ -89,8 +89,8 @@ internal static IReadOnlyList<U> SelectList<U>(IList<T> list, Func<T, U> f) {
         public int Count => data.Length;
 
         public IEnumerator<T> GetEnumerator() {
-            foreach (var e in data) {
-                yield return e;
+            for (int i = 0; i < data.Length; i++) {
+                yield return data[i];

```

Comment:
```suggestion
            return ((IEnumerable<T>)data).GetEnumerator();
```

Not sure if this is better.

Reply:
Probably is - fixed